crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/util/GenerateCsvUtilTest.java

```diff
@@ -0,0 +1,77 @@
+package uk.gov.justice.laa.crime.crowncourt.util;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.justice.laa.crime.crowncourt.entity.ReactivatedProsecutionCase;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.attribute.PosixFilePermission;
+import java.nio.file.attribute.PosixFilePermissions;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.anyList;
+import static org.mockito.ArgumentMatchers.anyString;
+
+@ExtendWith(MockitoExtension.class)
+class GenerateCsvUtilTest {
+
+
+    @Test
+    void testGenerateCsvFile() throws IOException {
+        // Create mock data
+        ReactivatedProsecutionCase case1 = ReactivatedProsecutionCase.builder()
+                .maatId(123)
+                .caseUrn("URN123")
+                .hearingId("HID123")
+                .previousOutcome("Outcome1")
+                .previousOutcomeDate(LocalDateTime.of(2024, 5, 5, 5, 5))
+                .dateOfStatusChange(LocalDateTime.of(2024, 6, 5, 5, 5)).build();
+        ReactivatedProsecutionCase case2 = ReactivatedProsecutionCase.builder()
+                .maatId(456)
+                .caseUrn("URN456")
+                .hearingId("HID456")
+                .previousOutcome("Outcome2")
+                .previousOutcomeDate(LocalDateTime.of(2024, 2, 1, 5, 5))
+                .dateOfStatusChange(LocalDateTime.of(2024, 6, 5, 5, 5)).build();
+
+        List<ReactivatedProsecutionCase> caseList = List.of(case1, case2);
+
+        // Call the method to test
+        File csvFile = GenerateCsvUtil.generateCsvFile(caseList, "testfile");
+
+        // Verify file exists
+        assertTrue(csvFile.exists(), "CSV file should exist");
+
+        // Verify file permissions
+        Set<PosixFilePermission> expectedPermissions = PosixFilePermissions.fromString("rwx------");
+        Set<PosixFilePermission> actualPermissions = Files.getPosixFilePermissions(csvFile.toPath());
+        assertEquals(expectedPermissions, actualPermissions, "File permissions should match");
+
+        // Verify file content
+        List<String> rows = Files.readAllLines(csvFile.toPath());
+        assertEquals(3, rows.size(), "CSV file should have three rows (header + 2 data rows)");
```

Comment:
I would consider grouping multiple test asserting in a JUnit `assertAll`

Reply:
we are ok with this.